lib/api/src/externals/function.rs

```diff
@@ -562,14 +574,70 @@ impl Function {
 
         match &self.definition {
             FunctionDefinition::Wasm(wasm) => {
-                self.call_wasm(&wasm, params, &mut results)?;
+                self.call_wasm(&wasm, params, &mut results, true)?;
             }
             _ => unimplemented!("The function definition isn't supported for the moment"),
         }
 
         Ok(results.into_boxed_slice())
     }
 
+    /// Call the `Function` function, without catching traps.
+    ///
+    /// This is a performance optimization for making function calls in batch.
+    /// It must be called within a callback passed to `wasmer::vm::catch_traps`.
+    ///
+    /// Depending on where the Function is defined, it will call it.
+    /// 1. If the function is defined inside a WebAssembly, it will call the trampoline
+    ///    for the function signature.
+    /// 2. If the function is defined in the host (in a native way), it will
+    ///    call the trampoline.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use wasmer::{imports, wat2wasm, Function, Instance, Module, Store, Type, Value, vm};
+    /// # let store = Store::default();
+    /// # let wasm_bytes = wat2wasm(r#"
+    /// # (module
+    /// #   (func (export "sum") (param $x i32) (param $y i32) (result i32)
+    /// #     local.get $x
+    /// #     local.get $y
+    /// #     i32.add
+    /// #   ))
+    /// # "#.as_bytes()).unwrap();
+    /// # let module = Module::new(&store, wasm_bytes).unwrap();
+    /// # let import_object = imports! {};
+    /// # let instance = Instance::new(&module, &import_object).unwrap();
+    /// #
+    /// let sum = instance.exports.get_function("sum").unwrap();
+    /// let vmctx = sum.get_vmctx();
+    /// unsafe {
+    ///     vm::catch_traps(vmctx, || {
+    ///         for i in 0..10 {
+    ///             assert_eq!(sum.call(&[Value::I32(1), Value::I32(2)]).unwrap().to_vec(), vec![Value::I32(3)]);
+    ///         }
+    ///     });
+    /// }
+    /// ```
+    pub fn call_unchecked(&self, params: &[Val]) -> Result<Box<[Val]>, RuntimeError> {
+        let mut results = vec![Val::null(); self.result_arity()];
+
+        match &self.definition {
+            FunctionDefinition::Wasm(wasm) => {
+                self.call_wasm(&wasm, params, &mut results, false)?;
+            }
+            _ => unimplemented!("The function definition isn't supported for the moment"),
+        }
+
+        Ok(results.into_boxed_slice())
+    }
+
+    /// Get a VM Context to manually handle traps when calling `call_unchecked`.
+    pub fn get_vmctx(&self) -> VMFunctionEnvironment {
```

Comment:
This function needs additional documentation about how to use it correctly.

It may also need to be `unsafe` depending on which functions exist that can take `VMFunctionEnvironment`.

`VMFunctionEnvironment` can either be a real `vmctx` or a host env, for `catch_traps` we must have a real `vmctx`. Which means this method must be called on an exported Wasm function.

This method doesn't strictly have to be unsafe if all other functions take a `VMFunctionEnvironment` are `unsafe` (which they should be); though the distinction between the types of `VMFunctionEnvironment` would be a useful thing to document here.

Reply:
Note: we shouldn't need this function once a future `catch_traps` refactor (using `&Store` instead of `vmctx`) lands